File: miranda/day27_ReverseBits.c
uint32_t reverseBits(uint32_t n) {
    uint32_t reversed = 0b0;

    for (int i = 0; i < 32; i++) {
        uint32_t bit = (n >> i) & 1;
        reversed = reversed | (bit << 31 - i);
    }

    return reversed;
}

File: miranda/day27_ReverseBits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        reversed = 0
        for i in range(32):
            bit = (n >> i) & 1
            reversed = reversed | (bit << 31 - i)
        return reversed
    
n = 0b11111111111111111111111111111101
solution = Solution()
print(solution.reverseBits(n))